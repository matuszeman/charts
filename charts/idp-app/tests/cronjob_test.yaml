suite: cronjob
templates:
  - cronjob.yaml
values:
  - values/minimum.yaml
tests:
  - it: when cronjob enabled
    values:
      - values/cronjob.yaml
    asserts:
      - matchSnapshot: { }
  - it: default restartPolicy
    values:
      - values/cronjob.yaml
    asserts:
      - equal:
          path: spec.jobTemplate.spec.template.spec.restartPolicy
          value: OnFailure
  - it: when restartPolicy is set
    values:
      - values/cronjob.yaml
    set:
      restartPolicy: Always
    asserts:
      - equal:
          path: spec.jobTemplate.spec.template.spec.restartPolicy
          value: Always
  - it: when spec params are set
    values:
      - values/cronjob.yaml
    set:
      cronjob:
        concurrencyPolicy: "Forbid"
        startingDeadlineSeconds: 30
        failedJobsHistoryLimit: 10
        successfulJobsHistoryLimit: 5
    asserts:
      - isSubset:
          path: spec
          content:
            concurrencyPolicy: "Forbid"
            startingDeadlineSeconds: 30
            failedJobsHistoryLimit: 10
            successfulJobsHistoryLimit: 5
  - it: when annotations are set
    values:
      - values/cronjob.yaml
    set:
      cronjob:
        annotations:
          some: annot
    asserts:
      - isSubset:
          path: metadata.annotations
          content:
            some: annot