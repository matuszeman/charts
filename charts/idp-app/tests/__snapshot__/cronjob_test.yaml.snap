when cronjob enabled:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      concurrencyPolicy: Allow
      failedJobsHistoryLimit: 3
      jobTemplate:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: app-name
            app.kubernetes.io/version: 0.1.0
        spec:
          template:
            metadata:
              labels:
                app.kubernetes.io/instance: RELEASE-NAME
                app.kubernetes.io/name: app-name
            spec:
              containers:
                - image: 123456.dkr.ecr.eu-central-1.amazonaws.com/test-job:0.1.0
                  imagePullPolicy: IfNotPresent
                  livenessProbe:
                    httpGet:
                      path: /
                      port: app
                  name: app
                  ports:
                    - containerPort: 80
                      name: app
                      protocol: TCP
                  readinessProbe:
                    httpGet:
                      path: /
                      port: app
                  resources:
                    limits:
                      memory: 256Mi
                    requests:
                      cpu: 100m
                      memory: 128Mi
              enableServiceLinks: false
              nodeSelector: null
              restartPolicy: OnFailure
              securityContext:
                sysctls:
                  - name: net.ipv4.ip_unprivileged_port_start
                    value: "0"
              serviceAccountName: RELEASE-NAME
              tolerations: null
              topologySpreadConstraints:
                - labelSelector:
                    matchLabels:
                      app.kubernetes.io/instance: RELEASE-NAME
                      app.kubernetes.io/name: app-name
                  maxSkew: 1
                  topologyKey: topology.kubernetes.io/zone
                  whenUnsatisfiable: ScheduleAnyway
                - labelSelector:
                    matchLabels:
                      app.kubernetes.io/instance: RELEASE-NAME
                      app.kubernetes.io/name: app-name
                  maxSkew: 1
                  topologyKey: kubernetes.io/hostname
                  whenUnsatisfiable: ScheduleAnyway
              volumes: null
      schedule: '@daily'
      successfulJobsHistoryLimit: 3
